week02/20.cpp

```diff
@@ -0,0 +1,74 @@
+class Solution
+{
+public:
+    bool isValid(string s)
+    {
+        stack<char> st;
+        char topStChar = '*';
+        for (int i = 0; i < s.length(); i++)
+        {
+            switch (s[i])
+            {
+            case '(':
+                st.push(s[i]);
+                break;
+            case '{':
+                st.push(s[i]);
+                break;
+            case '[':
+                st.push(s[i]);
+                break;
+            case ')':
+                if (st.empty())
+                {
+                    return false;
+                }
+                topStChar = st.top();
+                if (topStChar == '(')
+                {
+                    st.pop();
+                }
+                else
+                {
+                    return false;
+                }
+                break;
+            case '}':
+                if (st.empty())
+                {
+                    return false;
+                }
+                topStChar = st.top();
+                if (topStChar == '{')
+                {
+                    st.pop();
+                }
+                else
+                {
+                    return false;
+                }
+                break;
+            case ']':
+                if (st.empty())
+                {
+                    return false;
+                }
+                topStChar = st.top();
+                if (topStChar == '[')
+                {
+                    st.pop();
+                }
+                else
+                {
+                    return false;
+                }
+                break;
+            }
+        }
+        if (st.empty())
+        {
+            return true;
+        }
+        return false;
+    }
+};
```

Comment:
Again, huge code!

You won't get that much time in a live interview. Kindly refer to the [editorial](https://docs.google.com/document/d/1VipVcGyqiUDvFWbJ_yivTJucBRw3ct7HGZGGx-_CRsk/edit).